### PR TITLE
Add charity selection

### LINKED ISSUE #85: Charity selection for interest form.

### PR DESCRIPTION
**Pull request by bharathcs**
Fixes #85

I've added logos to the example.ts that you can use if you wanna test images.

https://user-images.githubusercontent.com/7360964/195632525-71a294f4-cf46-4f16-96c2-4529d9b72997.mov

Auto complete works too:
https://user-images.githubusercontent.com/7360964/195633223-8bb9e8a6-d1a6-4d34-90f3-f2d7bb3483e0.mp4

Removing all charities also raises error:
image

### CODE REVIEW

File: frontend/components/interests/form/InterestForm.tsx [hunk 1]
```diff
@@ -79,6 +79,12 @@ export default function InterestForm({ onSubmit }: InterestFormProps) {
                   Your Campaign
                 </Typography>
 
+                <InterestFormCharitySelector
+                  name="charities"
+                  label="Charities supported"
+                  placeholder="Choose your campaign beneficiaries."
+                />
+
                 <InterestFormTextInput name="campaignName" label="Name" placeholder="Give your campaign a name." />
```
Comment (sivayogasubramanian) [inline frontend/components/interests/form/InterestForm.tsx:88]:
```suggestion
                <InterestFormTextInput name="campaignName" label="Campaign Name" placeholder="Give your campaign a name." />
```

One more thing, when I tested it, it was giving me name (siva, bharath, etc) autocomplete suggestions. May want to turn that off.

**Review by sivayogasubramanian — COMMENTED** [review]
One suggestion. Will leave the rest of the review to Geng Ning.

File: frontend/components/generic/LogoBase64.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  alt?: string;
+  imgProps?: React.ImgHTMLAttributes<HTMLImageElement>;
+  sx?: SxProps<Theme>;
+}
+
+export default function LogoBase64({ base64Src, alt, imgProps, sx }: Props) {
+  imgProps = { ...imgProps, alt: imgProps?.alt ?? alt };
+  return (
+    <Box sx={sx}>
+      <img {...imgProps} src={`data:image/png;base64,${base64Src}`}></img>
```
Comment (sivayogasubramanian) [inline frontend/components/generic/LogoBase64.tsx:17]: Remove this encoding. This should be handled by the backend.

File: frontend/components/interests/form/InterestFormCharitySelector.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+const InterestFormCharitySelector = ({ name, label, placeholder }: Props) => {
+  const [, { error, touched }, { setValue, setTouched }] = useField<CharityMinimalData[]>(name);
+  // TODO: These are placeholders and will be removed once the charity API is up.
+  // Tracked in https://github.com/Giving-Coupons/giving-coupons/issues/132
+  const charities: CharityListData[] = [1, 2, 3, 4, 5].flatMap((v) => [
+    {
+      id: 1 * v,
+      name: 'Ark ' + v,
+      logoBase64:
```
Comment (sivayogasubramanian) [inline frontend/components/interests/form/InterestFormCharitySelector.tsx:23]: The format is wrong. See Geng Ning's example.ts

File: frontend/components/generic/LogoBase64.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Box, SxProps, Theme } from '@mui/material';
+
+interface Props {
+  // Base64 encoding of an image (prefix should not be included).
+  base64Src: string;
```
Comment (sivayogasubramanian) [inline frontend/components/generic/LogoBase64.tsx:5]:
```suggestion
  base64Src: Base64String;
```

To update after my PR is merged.

File: frontend/components/interests/form/InterestForm.tsx (hunk 1 above)
Comment (bharathcs) [reply frontend/components/interests/form/InterestForm.tsx:88]: Tracked in #135

File: frontend/components/generic/LogoBase64.tsx (hunk 4 above)
Comment (bharathcs) [reply frontend/components/generic/LogoBase64.tsx]: not relevant anymore since we're using

**Review by zognin — APPROVED** [review]
some nits but LGTM thanks

Are the logos meant to not be there yet (I'm assuming yes because the charities model doesn't have logoBase64 yet?)
Screenshot 2022-10-14 at 2 13 47 AM

**Pull request merged**